purchases/src/main/kotlin/com/revenuecat/purchases/listenerConversions.kt

```diff
@@ -172,13 +161,19 @@ fun Purchases.getOfferingsWith(
  * @param [onSuccess] Will be called after the purchase has completed
  * @param [onError] Will be called after the purchase has completed with error
  */
+@Deprecated(
+    message = "SkuDetails parameter replaced with StoreProduct. The callback now returns a StoreProduct.",
+    replaceWith = ReplaceWith(
+        "purchaseProductWith(activity, storeProduct, onError, onSuccess)"
+    )
+)
 fun Purchases.purchaseProductWith(
     activity: Activity,
     skuDetails: SkuDetails,
     onError: PurchaseErrorFunction = ON_PURCHASE_ERROR_STUB,
-    onSuccess: PurchaseCompletedFunction
+    onSuccess: DeprecatedPurchaseCompletedFunction
```

Comment:
this will cause a build error right? should we make the deprecated annotation an error then?

Reply:
Why would it cause a build error? The CustomerInfo rename?

Reply:
the onSuccess type change

Reply:
I see... I will try it out and see how it looks right now 😄